### PR TITLE
Fix Portal user info sample

### PR DESCRIPTION
**Pull request by har13205**
# Description

This PR is to fix OAuth window dismiss functionality in `Portal user info` sample. The window can be closed now and a button is added to authenticate portal to retry login after closing

## Type of change

- [x] Bug fix
- [ ] New sample implementation
- [ ] Sample viewer enhancement
- [ ] Other enhancement

**Platforms tested on**:

- [x] Windows
- [x] Android
- [ ] Linux
- [ ] macOS
- [ ] iOS

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/CloudAndPortal/PortalUserInfo/PortalUserInfo.cpp [hunk 1]
```diff
@@ -207,7 +215,9 @@ void PortalUserInfo::onPortalLoadStatusChanged(LoadStatus loadStatus)
     case LoadStatus::Loading:
         break;
     case LoadStatus::FailedToLoad:
-        if (m_portal)
+        if (m_portal->loadError().message() == "Code unauthorized.")
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/CloudAndPortal/PortalUserInfo/PortalUserInfo.cpp:218]: The previous code had a null check, which makes me wonder if your code still needs one.

File: ArcGISRuntimeSDKQt_CppSamples/CloudAndPortal/PortalUserInfo/PortalUserInfo.h [hunk 2]
```diff
@@ -62,6 +64,7 @@ class PortalUserInfo : public QQuickItem
   void accessChanged();
   void thumbnailUrlChanged();
   void loadErrorMessageChanged();
+  void portalClosed();
```
Comment (JamesMBallard) [inline ArcGISRuntimeSDKQt_CppSamples/CloudAndPortal/PortalUserInfo/PortalUserInfo.h:67]: Can you help me understand what this means? There is no concept of "closed" with Portal, so I think it means something else.

**Comment by har13205** [comment]
Done changes as per suggestions. By `closed` I meant about dismissing login window. Updated the name accordingly

**Review by ldanzinger — APPROVED** [review]
LGTM - I tested locally on Mac

**Review by JamesMBallard — APPROVED** [review]

**Pull request merged**